src/vanilla/utils/watch.ts

```diff
@@ -2,7 +2,9 @@ import { subscribe } from '../../vanilla.ts'
 
 type Cleanup = () => void
 type WatchGet = <T extends object>(proxyObject: T) => T
-type WatchCallback = (get: WatchGet) => Cleanup | void | undefined
+type WatchCallback = (
+  get: WatchGet,
+) => Cleanup | void | Promise<void | Cleanup> | undefined
```

Comment:
Just a tiny preference:
```suggestion
) => Cleanup | void | Promise<Cleanup | void> | undefined
```

Reply:
Done